// @aws-sdk/client-s3-control/dist-es/models/models_1.js
import { S3ControlServiceException as __BaseException } from "./S3ControlServiceException";
export const MFADelete = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
export class TooManyTagsException extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyTagsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyTagsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyTagsException.prototype);
        this.Message = opts.Message;
    }
}
export class JobStatusException extends __BaseException {
    constructor(opts) {
        super({
            name: "JobStatusException",
            $fault: "client",
            ...opts,
        });
        this.name = "JobStatusException";
        this.$fault = "client";
        Object.setPrototypeOf(this, JobStatusException.prototype);
        this.Message = opts.Message;
    }
}
export const RequestedJobStatus = {
    Cancelled: "Cancelled",
    Ready: "Ready",
};
