// @aws-sdk/client-s3-control/dist-cjs/models/models_1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequestedJobStatus = exports.JobStatusException = exports.TooManyTagsException = exports.MFADelete = void 0;
const S3ControlServiceException_1 = require("./S3ControlServiceException");
exports.MFADelete = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
class TooManyTagsException extends S3ControlServiceException_1.S3ControlServiceException {
    constructor(opts) {
        super({
            name: "TooManyTagsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyTagsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyTagsException.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyTagsException = TooManyTagsException;
class JobStatusException extends S3ControlServiceException_1.S3ControlServiceException {
    constructor(opts) {
        super({
            name: "JobStatusException",
            $fault: "client",
            ...opts,
        });
        this.name = "JobStatusException";
        this.$fault = "client";
        Object.setPrototypeOf(this, JobStatusException.prototype);
        this.Message = opts.Message;
    }
}
exports.JobStatusException = JobStatusException;
exports.RequestedJobStatus = {
    Cancelled: "Cancelled",
    Ready: "Ready",
};
