// @aws-sdk/middleware-sdk-s3-control/dist-cjs/host-prefix-deduplication/hostPrefixDeduplicationMiddleware.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getHostPrefixDeduplicationPlugin = exports.hostPrefixDeduplicationMiddlewareOptions = exports.hostPrefixDeduplicationMiddleware = void 0;
const deduplicateHostPrefix_1 = require("./deduplicateHostPrefix");
const hostPrefixDeduplicationMiddleware = () => {
    return (next, context) => async (args) => {
        var _a;
        const httpRequest = ((_a = args.request) !== null && _a !== void 0 ? _a : {});
        if (httpRequest === null || httpRequest === void 0 ? void 0 : httpRequest.hostname) {
            httpRequest.hostname = (0, deduplicateHostPrefix_1.deduplicateHostPrefix)(httpRequest.hostname);
        }
        return next(args);
    };
};
exports.hostPrefixDeduplicationMiddleware = hostPrefixDeduplicationMiddleware;
exports.hostPrefixDeduplicationMiddlewareOptions = {
    tags: ["HOST_PREFIX_DEDUPLICATION", "ENDPOINT_V2", "ENDPOINT"],
    toMiddleware: "serializerMiddleware",
    relation: "after",
    name: "hostPrefixDeduplicationMiddleware",
    override: true,
};
const getHostPrefixDeduplicationPlugin = (config) => ({
    applyToStack: (clientStack) => {
        clientStack.addRelativeTo((0, exports.hostPrefixDeduplicationMiddleware)(), exports.hostPrefixDeduplicationMiddlewareOptions);
    },
});
exports.getHostPrefixDeduplicationPlugin = getHostPrefixDeduplicationPlugin;
