// @aws-sdk/client-s3-control/dist-cjs/S3Control.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.S3Control = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const AssociateAccessGrantsIdentityCenterCommand_1 = require("./commands/AssociateAccessGrantsIdentityCenterCommand");
const CreateAccessGrantCommand_1 = require("./commands/CreateAccessGrantCommand");
const CreateAccessGrantsInstanceCommand_1 = require("./commands/CreateAccessGrantsInstanceCommand");
const CreateAccessGrantsLocationCommand_1 = require("./commands/CreateAccessGrantsLocationCommand");
const CreateAccessPointCommand_1 = require("./commands/CreateAccessPointCommand");
const CreateAccessPointForObjectLambdaCommand_1 = require("./commands/CreateAccessPointForObjectLambdaCommand");
const CreateBucketCommand_1 = require("./commands/CreateBucketCommand");
const CreateJobCommand_1 = require("./commands/CreateJobCommand");
const CreateMultiRegionAccessPointCommand_1 = require("./commands/CreateMultiRegionAccessPointCommand");
const CreateStorageLensGroupCommand_1 = require("./commands/CreateStorageLensGroupCommand");
const DeleteAccessGrantCommand_1 = require("./commands/DeleteAccessGrantCommand");
const DeleteAccessGrantsInstanceCommand_1 = require("./commands/DeleteAccessGrantsInstanceCommand");
const DeleteAccessGrantsInstanceResourcePolicyCommand_1 = require("./commands/DeleteAccessGrantsInstanceResourcePolicyCommand");
const DeleteAccessGrantsLocationCommand_1 = require("./commands/DeleteAccessGrantsLocationCommand");
const DeleteAccessPointCommand_1 = require("./commands/DeleteAccessPointCommand");
const DeleteAccessPointForObjectLambdaCommand_1 = require("./commands/DeleteAccessPointForObjectLambdaCommand");
const DeleteAccessPointPolicyCommand_1 = require("./commands/DeleteAccessPointPolicyCommand");
const DeleteAccessPointPolicyForObjectLambdaCommand_1 = require("./commands/DeleteAccessPointPolicyForObjectLambdaCommand");
const DeleteBucketCommand_1 = require("./commands/DeleteBucketCommand");
const DeleteBucketLifecycleConfigurationCommand_1 = require("./commands/DeleteBucketLifecycleConfigurationCommand");
const DeleteBucketPolicyCommand_1 = require("./commands/DeleteBucketPolicyCommand");
const DeleteBucketReplicationCommand_1 = require("./commands/DeleteBucketReplicationCommand");
const DeleteBucketTaggingCommand_1 = require("./commands/DeleteBucketTaggingCommand");
const DeleteJobTaggingCommand_1 = require("./commands/DeleteJobTaggingCommand");
const DeleteMultiRegionAccessPointCommand_1 = require("./commands/DeleteMultiRegionAccessPointCommand");
const DeletePublicAccessBlockCommand_1 = require("./commands/DeletePublicAccessBlockCommand");
const DeleteStorageLensConfigurationCommand_1 = require("./commands/DeleteStorageLensConfigurationCommand");
const DeleteStorageLensConfigurationTaggingCommand_1 = require("./commands/DeleteStorageLensConfigurationTaggingCommand");
const DeleteStorageLensGroupCommand_1 = require("./commands/DeleteStorageLensGroupCommand");
const DescribeJobCommand_1 = require("./commands/DescribeJobCommand");
const DescribeMultiRegionAccessPointOperationCommand_1 = require("./commands/DescribeMultiRegionAccessPointOperationCommand");
const DissociateAccessGrantsIdentityCenterCommand_1 = require("./commands/DissociateAccessGrantsIdentityCenterCommand");
const GetAccessGrantCommand_1 = require("./commands/GetAccessGrantCommand");
const GetAccessGrantsInstanceCommand_1 = require("./commands/GetAccessGrantsInstanceCommand");
const GetAccessGrantsInstanceForPrefixCommand_1 = require("./commands/GetAccessGrantsInstanceForPrefixCommand");
const GetAccessGrantsInstanceResourcePolicyCommand_1 = require("./commands/GetAccessGrantsInstanceResourcePolicyCommand");
const GetAccessGrantsLocationCommand_1 = require("./commands/GetAccessGrantsLocationCommand");
const GetAccessPointCommand_1 = require("./commands/GetAccessPointCommand");
const GetAccessPointConfigurationForObjectLambdaCommand_1 = require("./commands/GetAccessPointConfigurationForObjectLambdaCommand");
const GetAccessPointForObjectLambdaCommand_1 = require("./commands/GetAccessPointForObjectLambdaCommand");
const GetAccessPointPolicyCommand_1 = require("./commands/GetAccessPointPolicyCommand");
const GetAccessPointPolicyForObjectLambdaCommand_1 = require("./commands/GetAccessPointPolicyForObjectLambdaCommand");
const GetAccessPointPolicyStatusCommand_1 = require("./commands/GetAccessPointPolicyStatusCommand");
const GetAccessPointPolicyStatusForObjectLambdaCommand_1 = require("./commands/GetAccessPointPolicyStatusForObjectLambdaCommand");
const GetBucketCommand_1 = require("./commands/GetBucketCommand");
const GetBucketLifecycleConfigurationCommand_1 = require("./commands/GetBucketLifecycleConfigurationCommand");
const GetBucketPolicyCommand_1 = require("./commands/GetBucketPolicyCommand");
const GetBucketReplicationCommand_1 = require("./commands/GetBucketReplicationCommand");
const GetBucketTaggingCommand_1 = require("./commands/GetBucketTaggingCommand");
const GetBucketVersioningCommand_1 = require("./commands/GetBucketVersioningCommand");
const GetDataAccessCommand_1 = require("./commands/GetDataAccessCommand");
const GetJobTaggingCommand_1 = require("./commands/GetJobTaggingCommand");
const GetMultiRegionAccessPointCommand_1 = require("./commands/GetMultiRegionAccessPointCommand");
const GetMultiRegionAccessPointPolicyCommand_1 = require("./commands/GetMultiRegionAccessPointPolicyCommand");
const GetMultiRegionAccessPointPolicyStatusCommand_1 = require("./commands/GetMultiRegionAccessPointPolicyStatusCommand");
const GetMultiRegionAccessPointRoutesCommand_1 = require("./commands/GetMultiRegionAccessPointRoutesCommand");
const GetPublicAccessBlockCommand_1 = require("./commands/GetPublicAccessBlockCommand");
const GetStorageLensConfigurationCommand_1 = require("./commands/GetStorageLensConfigurationCommand");
const GetStorageLensConfigurationTaggingCommand_1 = require("./commands/GetStorageLensConfigurationTaggingCommand");
const GetStorageLensGroupCommand_1 = require("./commands/GetStorageLensGroupCommand");
const ListAccessGrantsCommand_1 = require("./commands/ListAccessGrantsCommand");
const ListAccessGrantsInstancesCommand_1 = require("./commands/ListAccessGrantsInstancesCommand");
const ListAccessGrantsLocationsCommand_1 = require("./commands/ListAccessGrantsLocationsCommand");
const ListAccessPointsCommand_1 = require("./commands/ListAccessPointsCommand");
const ListAccessPointsForObjectLambdaCommand_1 = require("./commands/ListAccessPointsForObjectLambdaCommand");
const ListJobsCommand_1 = require("./commands/ListJobsCommand");
const ListMultiRegionAccessPointsCommand_1 = require("./commands/ListMultiRegionAccessPointsCommand");
const ListRegionalBucketsCommand_1 = require("./commands/ListRegionalBucketsCommand");
const ListStorageLensConfigurationsCommand_1 = require("./commands/ListStorageLensConfigurationsCommand");
const ListStorageLensGroupsCommand_1 = require("./commands/ListStorageLensGroupsCommand");
const ListTagsForResourceCommand_1 = require("./commands/ListTagsForResourceCommand");
const PutAccessGrantsInstanceResourcePolicyCommand_1 = require("./commands/PutAccessGrantsInstanceResourcePolicyCommand");
const PutAccessPointConfigurationForObjectLambdaCommand_1 = require("./commands/PutAccessPointConfigurationForObjectLambdaCommand");
const PutAccessPointPolicyCommand_1 = require("./commands/PutAccessPointPolicyCommand");
const PutAccessPointPolicyForObjectLambdaCommand_1 = require("./commands/PutAccessPointPolicyForObjectLambdaCommand");
const PutBucketLifecycleConfigurationCommand_1 = require("./commands/PutBucketLifecycleConfigurationCommand");
const PutBucketPolicyCommand_1 = require("./commands/PutBucketPolicyCommand");
const PutBucketReplicationCommand_1 = require("./commands/PutBucketReplicationCommand");
const PutBucketTaggingCommand_1 = require("./commands/PutBucketTaggingCommand");
const PutBucketVersioningCommand_1 = require("./commands/PutBucketVersioningCommand");
const PutJobTaggingCommand_1 = require("./commands/PutJobTaggingCommand");
const PutMultiRegionAccessPointPolicyCommand_1 = require("./commands/PutMultiRegionAccessPointPolicyCommand");
const PutPublicAccessBlockCommand_1 = require("./commands/PutPublicAccessBlockCommand");
const PutStorageLensConfigurationCommand_1 = require("./commands/PutStorageLensConfigurationCommand");
const PutStorageLensConfigurationTaggingCommand_1 = require("./commands/PutStorageLensConfigurationTaggingCommand");
const SubmitMultiRegionAccessPointRoutesCommand_1 = require("./commands/SubmitMultiRegionAccessPointRoutesCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
const UpdateAccessGrantsLocationCommand_1 = require("./commands/UpdateAccessGrantsLocationCommand");
const UpdateJobPriorityCommand_1 = require("./commands/UpdateJobPriorityCommand");
const UpdateJobStatusCommand_1 = require("./commands/UpdateJobStatusCommand");
const UpdateStorageLensGroupCommand_1 = require("./commands/UpdateStorageLensGroupCommand");
const S3ControlClient_1 = require("./S3ControlClient");
const commands = {
    AssociateAccessGrantsIdentityCenterCommand: AssociateAccessGrantsIdentityCenterCommand_1.AssociateAccessGrantsIdentityCenterCommand,
    CreateAccessGrantCommand: CreateAccessGrantCommand_1.CreateAccessGrantCommand,
    CreateAccessGrantsInstanceCommand: CreateAccessGrantsInstanceCommand_1.CreateAccessGrantsInstanceCommand,
    CreateAccessGrantsLocationCommand: CreateAccessGrantsLocationCommand_1.CreateAccessGrantsLocationCommand,
    CreateAccessPointCommand: CreateAccessPointCommand_1.CreateAccessPointCommand,
    CreateAccessPointForObjectLambdaCommand: CreateAccessPointForObjectLambdaCommand_1.CreateAccessPointForObjectLambdaCommand,
    CreateBucketCommand: CreateBucketCommand_1.CreateBucketCommand,
    CreateJobCommand: CreateJobCommand_1.CreateJobCommand,
    CreateMultiRegionAccessPointCommand: CreateMultiRegionAccessPointCommand_1.CreateMultiRegionAccessPointCommand,
    CreateStorageLensGroupCommand: CreateStorageLensGroupCommand_1.CreateStorageLensGroupCommand,
    DeleteAccessGrantCommand: DeleteAccessGrantCommand_1.DeleteAccessGrantCommand,
    DeleteAccessGrantsInstanceCommand: DeleteAccessGrantsInstanceCommand_1.DeleteAccessGrantsInstanceCommand,
    DeleteAccessGrantsInstanceResourcePolicyCommand: DeleteAccessGrantsInstanceResourcePolicyCommand_1.DeleteAccessGrantsInstanceResourcePolicyCommand,
    DeleteAccessGrantsLocationCommand: DeleteAccessGrantsLocationCommand_1.DeleteAccessGrantsLocationCommand,
    DeleteAccessPointCommand: DeleteAccessPointCommand_1.DeleteAccessPointCommand,
    DeleteAccessPointForObjectLambdaCommand: DeleteAccessPointForObjectLambdaCommand_1.DeleteAccessPointForObjectLambdaCommand,
    DeleteAccessPointPolicyCommand: DeleteAccessPointPolicyCommand_1.DeleteAccessPointPolicyCommand,
    DeleteAccessPointPolicyForObjectLambdaCommand: DeleteAccessPointPolicyForObjectLambdaCommand_1.DeleteAccessPointPolicyForObjectLambdaCommand,
    DeleteBucketCommand: DeleteBucketCommand_1.DeleteBucketCommand,
    DeleteBucketLifecycleConfigurationCommand: DeleteBucketLifecycleConfigurationCommand_1.DeleteBucketLifecycleConfigurationCommand,
    DeleteBucketPolicyCommand: DeleteBucketPolicyCommand_1.DeleteBucketPolicyCommand,
    DeleteBucketReplicationCommand: DeleteBucketReplicationCommand_1.DeleteBucketReplicationCommand,
    DeleteBucketTaggingCommand: DeleteBucketTaggingCommand_1.DeleteBucketTaggingCommand,
    DeleteJobTaggingCommand: DeleteJobTaggingCommand_1.DeleteJobTaggingCommand,
    DeleteMultiRegionAccessPointCommand: DeleteMultiRegionAccessPointCommand_1.DeleteMultiRegionAccessPointCommand,
    DeletePublicAccessBlockCommand: DeletePublicAccessBlockCommand_1.DeletePublicAccessBlockCommand,
    DeleteStorageLensConfigurationCommand: DeleteStorageLensConfigurationCommand_1.DeleteStorageLensConfigurationCommand,
    DeleteStorageLensConfigurationTaggingCommand: DeleteStorageLensConfigurationTaggingCommand_1.DeleteStorageLensConfigurationTaggingCommand,
    DeleteStorageLensGroupCommand: DeleteStorageLensGroupCommand_1.DeleteStorageLensGroupCommand,
    DescribeJobCommand: DescribeJobCommand_1.DescribeJobCommand,
    DescribeMultiRegionAccessPointOperationCommand: DescribeMultiRegionAccessPointOperationCommand_1.DescribeMultiRegionAccessPointOperationCommand,
    DissociateAccessGrantsIdentityCenterCommand: DissociateAccessGrantsIdentityCenterCommand_1.DissociateAccessGrantsIdentityCenterCommand,
    GetAccessGrantCommand: GetAccessGrantCommand_1.GetAccessGrantCommand,
    GetAccessGrantsInstanceCommand: GetAccessGrantsInstanceCommand_1.GetAccessGrantsInstanceCommand,
    GetAccessGrantsInstanceForPrefixCommand: GetAccessGrantsInstanceForPrefixCommand_1.GetAccessGrantsInstanceForPrefixCommand,
    GetAccessGrantsInstanceResourcePolicyCommand: GetAccessGrantsInstanceResourcePolicyCommand_1.GetAccessGrantsInstanceResourcePolicyCommand,
    GetAccessGrantsLocationCommand: GetAccessGrantsLocationCommand_1.GetAccessGrantsLocationCommand,
    GetAccessPointCommand: GetAccessPointCommand_1.GetAccessPointCommand,
    GetAccessPointConfigurationForObjectLambdaCommand: GetAccessPointConfigurationForObjectLambdaCommand_1.GetAccessPointConfigurationForObjectLambdaCommand,
    GetAccessPointForObjectLambdaCommand: GetAccessPointForObjectLambdaCommand_1.GetAccessPointForObjectLambdaCommand,
    GetAccessPointPolicyCommand: GetAccessPointPolicyCommand_1.GetAccessPointPolicyCommand,
    GetAccessPointPolicyForObjectLambdaCommand: GetAccessPointPolicyForObjectLambdaCommand_1.GetAccessPointPolicyForObjectLambdaCommand,
    GetAccessPointPolicyStatusCommand: GetAccessPointPolicyStatusCommand_1.GetAccessPointPolicyStatusCommand,
    GetAccessPointPolicyStatusForObjectLambdaCommand: GetAccessPointPolicyStatusForObjectLambdaCommand_1.GetAccessPointPolicyStatusForObjectLambdaCommand,
    GetBucketCommand: GetBucketCommand_1.GetBucketCommand,
    GetBucketLifecycleConfigurationCommand: GetBucketLifecycleConfigurationCommand_1.GetBucketLifecycleConfigurationCommand,
    GetBucketPolicyCommand: GetBucketPolicyCommand_1.GetBucketPolicyCommand,
    GetBucketReplicationCommand: GetBucketReplicationCommand_1.GetBucketReplicationCommand,
    GetBucketTaggingCommand: GetBucketTaggingCommand_1.GetBucketTaggingCommand,
    GetBucketVersioningCommand: GetBucketVersioningCommand_1.GetBucketVersioningCommand,
    GetDataAccessCommand: GetDataAccessCommand_1.GetDataAccessCommand,
    GetJobTaggingCommand: GetJobTaggingCommand_1.GetJobTaggingCommand,
    GetMultiRegionAccessPointCommand: GetMultiRegionAccessPointCommand_1.GetMultiRegionAccessPointCommand,
    GetMultiRegionAccessPointPolicyCommand: GetMultiRegionAccessPointPolicyCommand_1.GetMultiRegionAccessPointPolicyCommand,
    GetMultiRegionAccessPointPolicyStatusCommand: GetMultiRegionAccessPointPolicyStatusCommand_1.GetMultiRegionAccessPointPolicyStatusCommand,
    GetMultiRegionAccessPointRoutesCommand: GetMultiRegionAccessPointRoutesCommand_1.GetMultiRegionAccessPointRoutesCommand,
    GetPublicAccessBlockCommand: GetPublicAccessBlockCommand_1.GetPublicAccessBlockCommand,
    GetStorageLensConfigurationCommand: GetStorageLensConfigurationCommand_1.GetStorageLensConfigurationCommand,
    GetStorageLensConfigurationTaggingCommand: GetStorageLensConfigurationTaggingCommand_1.GetStorageLensConfigurationTaggingCommand,
    GetStorageLensGroupCommand: GetStorageLensGroupCommand_1.GetStorageLensGroupCommand,
    ListAccessGrantsCommand: ListAccessGrantsCommand_1.ListAccessGrantsCommand,
    ListAccessGrantsInstancesCommand: ListAccessGrantsInstancesCommand_1.ListAccessGrantsInstancesCommand,
    ListAccessGrantsLocationsCommand: ListAccessGrantsLocationsCommand_1.ListAccessGrantsLocationsCommand,
    ListAccessPointsCommand: ListAccessPointsCommand_1.ListAccessPointsCommand,
    ListAccessPointsForObjectLambdaCommand: ListAccessPointsForObjectLambdaCommand_1.ListAccessPointsForObjectLambdaCommand,
    ListJobsCommand: ListJobsCommand_1.ListJobsCommand,
    ListMultiRegionAccessPointsCommand: ListMultiRegionAccessPointsCommand_1.ListMultiRegionAccessPointsCommand,
    ListRegionalBucketsCommand: ListRegionalBucketsCommand_1.ListRegionalBucketsCommand,
    ListStorageLensConfigurationsCommand: ListStorageLensConfigurationsCommand_1.ListStorageLensConfigurationsCommand,
    ListStorageLensGroupsCommand: ListStorageLensGroupsCommand_1.ListStorageLensGroupsCommand,
    ListTagsForResourceCommand: ListTagsForResourceCommand_1.ListTagsForResourceCommand,
    PutAccessGrantsInstanceResourcePolicyCommand: PutAccessGrantsInstanceResourcePolicyCommand_1.PutAccessGrantsInstanceResourcePolicyCommand,
    PutAccessPointConfigurationForObjectLambdaCommand: PutAccessPointConfigurationForObjectLambdaCommand_1.PutAccessPointConfigurationForObjectLambdaCommand,
    PutAccessPointPolicyCommand: PutAccessPointPolicyCommand_1.PutAccessPointPolicyCommand,
    PutAccessPointPolicyForObjectLambdaCommand: PutAccessPointPolicyForObjectLambdaCommand_1.PutAccessPointPolicyForObjectLambdaCommand,
    PutBucketLifecycleConfigurationCommand: PutBucketLifecycleConfigurationCommand_1.PutBucketLifecycleConfigurationCommand,
    PutBucketPolicyCommand: PutBucketPolicyCommand_1.PutBucketPolicyCommand,
    PutBucketReplicationCommand: PutBucketReplicationCommand_1.PutBucketReplicationCommand,
    PutBucketTaggingCommand: PutBucketTaggingCommand_1.PutBucketTaggingCommand,
    PutBucketVersioningCommand: PutBucketVersioningCommand_1.PutBucketVersioningCommand,
    PutJobTaggingCommand: PutJobTaggingCommand_1.PutJobTaggingCommand,
    PutMultiRegionAccessPointPolicyCommand: PutMultiRegionAccessPointPolicyCommand_1.PutMultiRegionAccessPointPolicyCommand,
    PutPublicAccessBlockCommand: PutPublicAccessBlockCommand_1.PutPublicAccessBlockCommand,
    PutStorageLensConfigurationCommand: PutStorageLensConfigurationCommand_1.PutStorageLensConfigurationCommand,
    PutStorageLensConfigurationTaggingCommand: PutStorageLensConfigurationTaggingCommand_1.PutStorageLensConfigurationTaggingCommand,
    SubmitMultiRegionAccessPointRoutesCommand: SubmitMultiRegionAccessPointRoutesCommand_1.SubmitMultiRegionAccessPointRoutesCommand,
    TagResourceCommand: TagResourceCommand_1.TagResourceCommand,
    UntagResourceCommand: UntagResourceCommand_1.UntagResourceCommand,
    UpdateAccessGrantsLocationCommand: UpdateAccessGrantsLocationCommand_1.UpdateAccessGrantsLocationCommand,
    UpdateJobPriorityCommand: UpdateJobPriorityCommand_1.UpdateJobPriorityCommand,
    UpdateJobStatusCommand: UpdateJobStatusCommand_1.UpdateJobStatusCommand,
    UpdateStorageLensGroupCommand: UpdateStorageLensGroupCommand_1.UpdateStorageLensGroupCommand,
};
class S3Control extends S3ControlClient_1.S3ControlClient {
}
exports.S3Control = S3Control;
(0, smithy_client_1.createAggregatedClient)(commands, S3Control);
