// @aws-sdk/client-s3-control/dist-es/commands/GetMultiRegionAccessPointPolicyStatusCommand.js
import { getProcessArnablesPlugin } from "@aws-sdk/middleware-sdk-s3-control";
import { getApplyMd5BodyChecksumPlugin } from "@smithy/middleware-apply-body-checksum";
import { getEndpointPlugin } from "@smithy/middleware-endpoint";
import { getSerdePlugin } from "@smithy/middleware-serde";
import { Command as $Command } from "@smithy/smithy-client";
import { SMITHY_CONTEXT_KEY, } from "@smithy/types";
import { de_GetMultiRegionAccessPointPolicyStatusCommand, se_GetMultiRegionAccessPointPolicyStatusCommand, } from "../protocols/Aws_restXml";
export { $Command };
export class GetMultiRegionAccessPointPolicyStatusCommand extends $Command {
    static getEndpointParameterInstructions() {
        return {
            RequiresAccountId: { type: "staticContextParams", value: true },
            AccountId: { type: "contextParams", name: "AccountId" },
            UseArnRegion: { type: "clientContextParams", name: "useArnRegion" },
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, GetMultiRegionAccessPointPolicyStatusCommand.getEndpointParameterInstructions()));
        this.middlewareStack.use(getProcessArnablesPlugin(configuration));
        this.middlewareStack.use(getApplyMd5BodyChecksumPlugin(configuration));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "S3ControlClient";
        const commandName = "GetMultiRegionAccessPointPolicyStatusCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: (_) => _,
            outputFilterSensitiveLog: (_) => _,
            [SMITHY_CONTEXT_KEY]: {
                service: "AWSS3ControlServiceV20180820",
                operation: "GetMultiRegionAccessPointPolicyStatus",
            },
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return se_GetMultiRegionAccessPointPolicyStatusCommand(input, context);
    }
    deserialize(output, context) {
        return de_GetMultiRegionAccessPointPolicyStatusCommand(output, context);
    }
}
